pkg/imgutil/snapshotter.go

```diff
@@ -77,9 +80,18 @@ func (rs *remoteSnapshotterOpts) isRemote() bool {
 
 func (rs *remoteSnapshotterOpts) apply(config *pull.Config, ref string) {
 	h := ctdsnapshotters.AppendInfoHandlerWrapper(ref)
+
+	if rs.snapshotter == "soci" {
+		h = socisource.AppendDefaultLabelsHandlerWrapper(
+			"", ctdsnapshotters.AppendInfoHandlerWrapper(ref))
+
+		config.RemoteOpts = append(config.RemoteOpts, containerd.WithPullUnpack)
```

Comment:
Why do we need `WithPullUnpack` here? It seems already appended here : https://github.com/containerd/nerdctl/blob/15e68a0cbc9d39e19aadab1a0860c2975bd54743/pkg/imgutil/imgutil.go#L229

Reply:
@CodeChanning If you could verify SOCI can work without "WithPullUnpack", you can also remove the test case of `unpackB = true`.

Reply:
I took out the extra appending of `WithPullUnpack` and took out the `unpackB = true` test case.